packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-field-auth-apikey.ts

```diff
@@ -0,0 +1,619 @@
+        }
+
+        # Api Key can't create a record with SSN value
+        # Api Key can create a record without a SSN value
+        # Api Key can't update a record with SSN value
+        # Api Key can't delete a record
+        type Person2 @model @auth(rules: [{allow: public, provider: apiKey}, {allow: public, provider: iam}]) {
+          id: Int! @primaryKey
+          firstName: String
+          lastName: String
+          ssn: String @auth(rules: [{allow: public, provider: iam}, {allow: public, provider: apiKey, operations: [read]}])
+        }
+
+        # Api Key can't create a record with SSN value
+        # Api Key can delete a record
+        type Person3 @model @auth(rules: [{allow: public, provider: apiKey}, {allow: public, provider: iam}]) {
+          id: Int! @primaryKey
+          firstName: String
+          lastName: String
+          ssn: String @auth(rules: [{allow: public, provider: iam}, {allow: public, provider: apiKey, operations: [delete]}])
+        }
+      `;
+      writeFileSync(rdsSchemaFilePath, schema, 'utf8');
+
+      // Enable unauthenticated access to the Cognito resource and push again
+      await enableUserPoolUnauthenticatedAccess(projRoot);
+    };
+
+    const selectionSetWithoutSSN = `
+      id
+      firstName
+      lastName
+    `;
+
+    const getQueryWithoutSSN = (name: string): string => /* GraphQL */ `
+      query Get${name}($id: Int!) {
+        get${name}(id: $id) {
+          ${selectionSetWithoutSSN}
+        }
+      }
+    `;
+
+    const listQueryWithoutSSN = (name: string): string => /* GraphQL */ `
+      query List${name}s {
+        list${name}s {
+          items {
+            ${selectionSetWithoutSSN}
+          }
+        }
+      }
+    `;
+
+    test('Model Person1 - Api Key can create a record with all fields, but SSN response is null', async () => {
+      const person = {
+        id: 1,
+        firstName: 'Person1',
+        lastName: 'Person1',
+        ssn: '123-45-6789',
+      };
+      // The API can create the record with all fields. But the API throws unauthorized error for the SSN field.
+      // We will run a get query to verify that the record was created with all the fields including SSN.
+      await expect(person1ApiKeyClient.create('createPerson1', person)).rejects.toThrow(
+        'GraphQL error: Not Authorized to access ssn on type Person1',
+      );
+
+      const person1Response = await person1IAMPublicClient.get({ id: person.id });
+      expect(person1Response.data.getPerson1.id).toEqual(person.id);
+      expect(person1Response.data.getPerson1.firstName).toEqual(person.firstName);
+      expect(person1Response.data.getPerson1.lastName).toEqual(person.lastName);
+      expect(person1Response.data.getPerson1.ssn).toEqual(person.ssn);
+    });
+
+    test('Model Person1 - Api Key cannot read SSN field on GET query', async () => {
+      const person = {
+        id: 1,
```

Comment:
nit: These tests share the same state with the previous tests, which could make it hard to debug.

Reply:
This is because we use the same model in different tests. In case of an issue, we need to fix the one failing at the top, then continue with others. For troubleshooting, we can easily rerun the tests by truncating the tables each time.